tests/telemetry/test_telemetry.py

```diff
@@ -164,29 +164,42 @@ def test_telemetry_queue_buffer_cnt(duthosts, enum_rand_one_per_hwsku_hostname,
     logger.info('start telemetry output testing')
     dut_ip = duthost.mgmt_ip
 
+    interfaces_status = duthost.show_and_parse("show interface status")
+    admin_up_interfaces = [item['interface'] for item in interfaces_status if item['admin'] == 'up']
+
     duthost.shell("sonic-cfggen -d --print-data > {}".format(ORIG_CFG_DB))
     data = json.loads(duthost.shell("cat {}".format(ORIG_CFG_DB),
                                     verbose=False)['stdout'])
+
     buffer_queues = list(data['BUFFER_QUEUE'].keys())
-    iface_to_check = buffer_queues[0].split('|')[0]
-    iface_buffer_queues = [bq for bq in buffer_queues if any(val in iface_to_check for val in bq.split('|'))]
+    buffer_queues_interfaces = [bq.split('|')[0] for bq in buffer_queues]
+
+    interface_to_check = None
+    for bq in buffer_queues_interfaces:
+        if bq in admin_up_interfaces:
+            interface_to_check = bq
+            break
+    if interface_to_check is None:
+        pytest.skip("Skipping test as there are none interfaces in admin'up' state with buffer queues to check")
```

Comment:
I think we should fail the test as to not hide the issue of all ports being down.

Reply:
This for the case there are ports in up state but no buffer queues configured as in that case there is no meaning of testing the buffer queues counters. Do you want me to fail the test instead?

Reply:
Could we add a check before to ensure that there are ports up before we check for those with buffer queues configured? And fail if there are no ports up at all?

Reply:
There is a sanity_check before each test to ensure all `admin_up` ports are up. So I think it's probably good to skip the test if no port is `admin_up`?